instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex

```diff
@@ -102,11 +102,21 @@ defmodule OpentelemetryEcto do
           acc
       end)
 
-    parent_context = OpentelemetryProcessPropagator.fetch_parent_ctx(1, :"$callers")
+    parent_context =
+      case OpentelemetryProcessPropagator.fetch_ctx(self()) do
+        :undefined ->
```

Comment:
This should just attempt to get the context directly and not go through the propagator.

```suggestion
      case OpenTelemetry.Ctx.get_current() do
        %{} ->
```

Reply:
Empty map matches any map, so this pattern can't distinguish between "no current ctx" from "any ctx". I could change to something like:

```suggestion
      case OpenTelemetry.Ctx.get_current() do
        ctx when map_size(ctx) == 0 ->
```

The approach of `OpentelemetryProcessPropagator.fetch_ctx` is just so the code that manipulate the ctx here is consistent (everything goes through the propagator). I can change the strategy to the proposed one if you prefer :)

Reply:
Maybe we should reference current span ctx instead since it will return undefined if there isn't a span in progress.

https://github.com/open-telemetry/opentelemetry-erlang/blob/main/apps/opentelemetry_api/src/otel_tracer.erl#L121

@tsloughter any thoughts here?

Reply:
what is it trying to do here?

Reply:
This instrumentation may run from inside Task. There is the case of preloads which spawn a Task and we want to find the context from $callers, to connect spans. If the query is not a preload, there is no intermediary Task.

In case of no intermediary Task (from preload), this code will skip a legit Task from client code. The solution is to search the current process first, to have an opportunity to short-circuit the problematic scenario.

Reply:
aah. But if its not overriding the context and uses the macros to start/end spans then it should already use any current context that might exist in the process. There should be no need to get the current context to attach it as far as I can understand it.